### DISCUSSION TITLE
GraphQL subscription in tanstack query

### DISCUSSION THREAD

**Discussion by trishagonzales** [TanStack/query#6258] [Q&A]
Hi, I'm new to react query and I'm trying to implement graphql subscription. There are examples and guides both in tanstack and the-guild.dev docs about graphql queries and mutations in react query but nothing for subscription.

I tried to connect sockets on useEffect and call setQueryData when receiving new updates but setQueryData needs a query key and a query key needs an observer. Is a query observer necessary when data is always updated directly in cache? If so, what do i put in queryFn?

```
const { data } = useQuery({
   queryKey: ['countdown']
});
```

**Comment by TkDodo** [comment]
> and a query key needs an observer

not sure what that means. have you read:

https://tkdodo.eu/blog/using-web-sockets-with-react-query

if you have no resource to query, meaning no endpoint to make the initial fetch from, don't use react-query - just setup the subscription and push data somewhere else :)
